src/lib.rs

```diff
@@ -29,7 +29,6 @@ pub use crate::{
     parse::{ArgMatches, Indices, OsValues, Values},
 };
 
-#[cfg(feature = "derive")]
 pub use crate::derive::{ArgEnum, Args, Clap, FromArgMatches, IntoApp, Subcommand};
```

Comment:
Let's put the `ArgEnum` in a separate file and then make it a normal import instead of removing the feature gate for all the traits in `derive.rs`.

Reply:
I intentionally exposed them all
- I think we should leave it open for people to hand implement these traits (I can see us doing it in `clap_generate`) so people can provide mixin arguments without forcing `derive`
- I don't normally see crates that have a `derive` feature flag put the related traits behind the feature flag as well.

Reply:
Part of my original idea was to expose all those traits by default, but allow others to implement them if they so choose. What became `clap_derive` was just one implementation of them provided by us. Similar to the `log` facade, we need to expose the traits themselves (but not their implementations) by default so that downstream crates can all be speaking the same language.